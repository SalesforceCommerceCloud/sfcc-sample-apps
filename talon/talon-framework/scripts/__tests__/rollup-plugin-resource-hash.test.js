import resourceHashPlugin from '../rollup-plugin-resource-hash';
import * as fs from 'fs';

jest.mock('fs');

jest.mock('talon-common', () => ({
    getResourceUrl: (name, mode, hash) => {
        return `${hash}/${name}/${mode}`;
    }
}));

jest.mock('rollup-plugin-hash', () => {
    return {
        __esModule: true,
        default: (opt) => {
            return {
                onwrite: (filename) => {
                    opt.callback(filename);
                }
            };
        },
    };
});

describe('rollup-plugin-resource-hash', () => {
    describe('onwrite', () => {
        it('writes out new resource map', async () => {
            const plugin = resourceHashPlugin({
                mode: 'dev',
                resourceDescriptor: 'talontest'
            });
            plugin.onwrite('testfilename');

            expect(fs.writeFileSync.mock.calls[0][1]).toMatchSnapshot();
        });
        it('adds to existing resource map', async () => {
            const existing = {
                "framework://talon": {
                    "dev": "aa6f1ead02"
                }
            };

            fs.existsSync.mockReturnValue(true);
            fs.readFileSync.mockReturnValue(JSON.stringify(existing));

            const plugin = resourceHashPlugin({
                mode: 'dev',
                resourceDescriptor: 'talontest'
            });
            plugin.onwrite('testfilename');

            const output = fs.writeFileSync.mock.calls[1][1];
            expect(JSON.parse(output)["framework://talon"]).toEqual(existing["framework://talon"]);
            expect(output).toMatchSnapshot();
        });
    });
});