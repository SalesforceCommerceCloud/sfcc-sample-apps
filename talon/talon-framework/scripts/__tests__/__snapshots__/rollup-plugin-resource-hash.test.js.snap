// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup-plugin-resource-hash onwrite adds to existing resource map 1`] = `
"{
    \\"framework://talon\\": {
        \\"dev\\": \\"aa6f1ead02\\"
    },
    \\"talontest\\": {
        \\"dev\\": \\"testfilename\\"
    }
}"
`;

exports[`rollup-plugin-resource-hash onwrite writes out new resource map 1`] = `
"{
    \\"talontest\\": {
        \\"dev\\": \\"testfilename\\"
    }
}"
`;
